dockerfile_parse/parser.py

```diff
@@ -880,7 +880,9 @@ def image_from(from_value):
         )?
         """)
     match = re.match(regex, from_value)
-    return match.group('image', 'name') if match else (None, None)
+    image = ImageName.parse(match.group('image')) if match else None
+    name = match.group('name') if match else None
+    return image, name
```

Comment:
we are breaking BW compatibility here, I'm not sure if we can do this

Reply:
As far as i'm aware it shouldn't break backwards compatibility:

Old version: Image was a string
New version: Image is the new ImageName class which has a str representation (which is idential to the original string of the previous version).

This should be backwards compatible (the unit tests reflect this).

Reply:
Addition:

Just realized what I typed above is not 100% true: See the explicit str(s) in util.py (line 51). 
impact should still be minimal.

Alternative solution would be keeping the old function.

Reply:
yes, explicit casting to str() is required, that's breaking change

Reply:
IMO would be safer to add new function and add deprecation warning to this one

Reply:
Done!
I could not find any examples within the containerbuildsystem project with deprecations so i opted to use the `deprecation` module. Let me know if you want to use a different standard.